src/Diagnostics.CompilerHost/Controllers/CompilerHostController.cs

```diff
@@ -29,24 +28,32 @@ public async Task<IActionResult> Post([FromBody]JToken jsonBody)
                 return BadRequest("Missing body");
             }
 
-            string script = jsonBody.Value<string>("script");
+            // Get script and reference
+            var script = jsonBody.Value<string>("script");
+
+            var references = new Dictionary<string, string>();
+            if (jsonBody["reference"] != null)
+            {
+                references = JsonConvert.DeserializeObject<Dictionary<string, string>>(jsonBody["reference"].ToString());
+            }
 
             if (string.IsNullOrWhiteSpace(script))
             {
                 return BadRequest("Missing script from body");
             }
 
-            EntityMetadata metaData = new EntityMetadata(script);
-            CompilerResponse compilerResponse = new CompilerResponse();
-            using (var invoker = new EntityInvoker(metaData, ScriptHelper.GetFrameworkReferences(), ScriptHelper.GetFrameworkImports()))
+            var metaData = new EntityMetadata(script);
+            var compilerResponse = new CompilerResponse();
+            using (var invoker = new EntityInvoker(metaData, ScriptHelper.GetFrameworkReferences(), ScriptHelper.GetFrameworkImports(), references.ToImmutableDictionary()))
             {
                 await invoker.InitializeEntryPointAsync();
                 compilerResponse.CompilationOutput = invoker.CompilationOutput;
                 compilerResponse.CompilationSucceeded = invoker.IsCompilationSuccessful;
+                compilerResponse.References = invoker.References;
```

Comment:
This is more of curiosity and a question, but how is the client using the references in the compilation response?

Reply:
The client will use the references to figure out which gists the detector(or gist) loads.